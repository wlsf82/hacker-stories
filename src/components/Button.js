import React from 'react';

export const Button = ({
  type = 'button',
  isDisabled = false,
  onClick,
  children
}) =>
  <button type={type} disabled={isDisabled} onClick={onClick}>
    {children}
  </button>;
