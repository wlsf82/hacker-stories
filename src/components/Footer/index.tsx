import React from 'react';

import './Footer.css';

const Footer = () => (
  <footer>
    Icons made by <a
      href="https://www.flaticon.com/authors/freepik"
      title="Freepik"
      target="_blank"
      rel="noopener noreferrer"
    >
      Freepik
    </a> from <a
      href="https://www.flaticon.com/"
      title="Flaticon"
      target="_blank"
      rel="noopener noreferrer"
    >
      www.flaticon.com
    </a>
  </footer>
)

export default Footer;
