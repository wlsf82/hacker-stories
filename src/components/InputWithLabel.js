import React from 'react';

export const InputWithLabel = ({
  id,
  value,
  type = 'text',
  onInputChange,
  isFocused,
  children,
}) => {
  const inputRef = React.useRef();

  React.useEffect(() => {
    if (isFocused && inputRef.current) {
      inputRef.current.focus();
    }
  }, [isFocused])

  return (
    <>
      <label htmlFor={id}>{children}</label>
      <input
        ref={inputRef}
        id={id}
        type={type}
        value={value}
        onChange={onInputChange}
      />
    </>
  )
}
