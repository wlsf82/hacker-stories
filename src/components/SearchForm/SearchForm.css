.search-form {
  padding: 10px 0 20px 0;
  display: flex;
  align-items: baseline;
}
