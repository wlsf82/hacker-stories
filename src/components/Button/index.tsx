import React from 'react';

import { ButtonProps } from '../../types';

import './Button.css';

const Button = ({
  type = 'button',
  isDisabled = false,
  onClick,
  className,
  children
}: ButtonProps) => (
  <button
    className={className}
    type={type}
    disabled={isDisabled}
    onClick={onClick}
  >
    {children}
  </button>
);

export default Button;
