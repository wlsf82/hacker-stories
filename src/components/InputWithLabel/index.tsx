import React from 'react';

import { InputWithLabelProps } from '../../types';

import './InputWithLabel.css';

export const InputWithLabel = ({
  id,
  value,
  type = 'text',
  onInputChange,
  isFocused,
  children,
}: InputWithLabelProps) => {
  const inputRef = React.useRef<HTMLInputElement>(null!);

  React.useEffect(() => {
    if (isFocused && inputRef.current) {
      inputRef.current.focus();
    }
  }, [isFocused]);

  return (
    <>
      <label htmlFor={id} className="label">{children}</label>
      &nbsp;
      <input
        ref={inputRef}
        id={id}
        type={type}
        value={value}
        onChange={onInputChange}
        className="input"
      />
    </>
  );
};
