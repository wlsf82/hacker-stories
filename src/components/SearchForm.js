import React from 'react';

import { Button } from './Button';
import { InputWithLabel } from './InputWithLabel';

export const SearchForm = ({
  searchTerm,
  onSearchInput,
  onSearchSubmit,
}) => (
  <form onSubmit={onSearchSubmit} className="search-form">
    <InputWithLabel
      id="search"
      value={searchTerm}
      isFocused
      onInputChange={onSearchInput}
    >
      <strong>Search:</strong>
    </InputWithLabel>

    <Button
      type="submit"
      isDisabled={!searchTerm}
      className="button button-large"
    >
      Submit
    </Button>
  </form>
);
