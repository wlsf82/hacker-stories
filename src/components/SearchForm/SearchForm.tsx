import React from 'react';

import { Button } from '../Button/Button';
import { InputWithLabel } from '../InputWithLabel/InputWithLabel';

import { SearchFormProps } from '../../types';

import './SearchForm.css';

export const SearchForm = ({
  searchTerm,
  onSearchInput,
  onSearchSubmit,
}: SearchFormProps) => (
  <form onSubmit={onSearchSubmit} className="search-form">
    <InputWithLabel
      id="search"
      value={searchTerm}
      isFocused
      onInputChange={onSearchInput}
    >
      <strong>Search:</strong>
    </InputWithLabel>

    <Button
      type="submit"
      isDisabled={!searchTerm}
      className="button button-large"
    >
      Submit
    </Button>
  </form>
);
