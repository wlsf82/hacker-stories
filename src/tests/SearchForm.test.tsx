import React from 'react';
import renderer from 'react-test-renderer';

import { SearchForm } from '../components/SearchForm';

describe('SearchForm', () => {
  const searchFormProps = {
    searchTerm: 'React',
    onSearchInput: jest.fn(),
    onSearchSubmit: jest.fn(),
  };

  let component: any;

  beforeEach(() => {
    component = renderer.create(<SearchForm {...searchFormProps} />);
  });

  it('renders the input field with its value', () => {
    const value = component.root.findByType('input').props.value;

    expect(value).toEqual(searchFormProps.searchTerm);
  });

  it('changes the input field', () => {
    const pseudoEvent = { target: 'Redux' };

    component.root.findByType('input').props.onChange(pseudoEvent);

    expect(searchFormProps.onSearchInput).toHaveBeenCalledTimes(1);
    expect(searchFormProps.onSearchInput).toHaveBeenCalledWith(pseudoEvent);
  });

  it('submits the form', () => {
    const pseudoEvent = {};

    component.root.findByType('form').props.onSubmit(pseudoEvent);

    expect(searchFormProps.onSearchSubmit).toHaveBeenCalledTimes(1);
    expect(searchFormProps.onSearchSubmit).toHaveBeenCalledWith(pseudoEvent);
  });

  it('disables the button and prevents submit', () => {
    component.update(<SearchForm {...searchFormProps} searchTerm="" />);

    expect(component.root.findByType('button').props.disabled).toBeTruthy();
  });
});
