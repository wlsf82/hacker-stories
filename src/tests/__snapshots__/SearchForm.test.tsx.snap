// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders snapshot 1`] = `
<form
  className="search-form"
  onSubmit={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {},
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <label
    className="label"
    htmlFor="search"
  >
    <strong>
      Search:
    </strong>
  </label>
  Â 
  <input
    className="input"
    id="search"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "target": "Redux",
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    type="text"
    value="React"
  />
  <button
    className="button button-large"
    disabled={false}
    type="submit"
  >
    Submit
  </button>
</form>
`;
